stdlib/src/utils/string_slice.mojo

```diff
@@ -31,6 +31,39 @@ alias StaticString = StringSlice[ImmutableStaticLifetime]
 """An immutable static string slice."""
 
 
+fn _unicode_codepoint_utf8_byte_length(c: Int) -> Int:
+    alias sizes = SIMD[DType.int32, 4](0, 0b0111_1111, 0b0111_1111_1111, 0xFFFF)
+    return int((sizes < c).cast[DType.uint8]().reduce_add())
+
+
+fn _shift_unicode_to_utf8(ptr: UnsafePointer[UInt8], c: Int, num_bytes: Int):
```

Comment:
I'm not sure of this argument ordering but I don't know if we have any such sort of convention yet

Reply:
There's no convention yet, I could make a case either way: weak preference towards putting `num_bytes` as the second argument.